### PR TITLE
PT-11555: Removed Obsolete attributes

### PR DESCRIPTION
**Pull request by OlegoO**
## Description
fix: Removed Obsolete attributes for some method to prevent from son)ar warning.
fix: Removed protected virtual async Task  ryAdjustOrderInventory(GenericChangedEntry<CustomerOrder> changedEntry).
feat: Update documentation for some methods.

### Jira-link:
https://virtocommerce.atlassian.net/browse/PT-11555

### CODE REVIEW

File: src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs [hunk 1]
```diff
@@ -11,25 +11,49 @@ namespace VirtoCommerce.OrdersModule.Core.Model
...
+        ///</summary>
         public string ChannelId { get; set; }
 
+        ///<summary>
+        /// Represents the unique identifier of the store where the order was placed, which could be a physical store, online store, or mobile app.
+        ///</summary>
         public string StoreId { get; set; }
 
+        ///<summary>
+        /// Represents the unique identifier of the store where the order was placed, which could be a physical store, online store, or mobile app.
```
Comment (artem-dudarev) [inline src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs:35]: Name of the store

File: src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs [hunk 2]
```diff
@@ -11,25 +11,49 @@ namespace VirtoCommerce.OrdersModule.Core.Model
...
+        ///</summary>
         public string StoreName { get; set; }
 
+        ///<summary>
+        /// Represents the unique identifier of the organization that owns the store where the order was placed.
```
Comment (artem-dudarev) [inline src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs:40]: Organization which the customer belongs to

File: src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs [hunk 3]
```diff
@@ -11,25 +11,49 @@ namespace VirtoCommerce.OrdersModule.Core.Model
...
+        ///</summary>
         public string OrganizationId { get; set; }
 
+        ///<summary>
+        /// Represents the name of the organization that owns the store where the order was placed.
```
Comment (artem-dudarev) [inline src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs:45]: Organization which the customer belongs to

File: src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs [hunk 4]
```diff
@@ -11,25 +11,49 @@ namespace VirtoCommerce.OrdersModule.Core.Model
...
+        ///</summary>
         public string OrganizationName { get; set; }
 
-        [Obsolete("Use CustomerId instead")]
+        ///<summary>
+        /// Represents the name of the organization that owns the store where the order was placed.
```
Comment (artem-dudarev) [inline src/VirtoCommerce.OrdersModule.Core/Model/CustomerOrder.cs:50]: Unique identifier of the employee

File: src/VirtoCommerce.OrdersModule.Core/Services/ICustomerOrderSearchService.cs [hunk 5]
```diff
@@ -10,7 +10,6 @@ namespace VirtoCommerce.OrdersModule.Core.Services
     /// </summary>
     public interface ICustomerOrderSearchService
```
Comment (OlegoO) [inline src/VirtoCommerce.OrdersModule.Core/Services/ICustomerOrderSearchService.cs:11]: Remove

**Pull request closed without merging**